src/renderer/components/NewProjectModal/UploadVideoView.tsx

```diff
@@ -0,0 +1,132 @@
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Box, Stack, styled } from '@mui/material';
+import colors from 'renderer/colors';
+import CloseIcon from '@mui/icons-material/Close';
+import { ApplicationStore } from 'renderer/store/helpers';
+import { projectCreated, recentProjectAdded } from 'renderer/store/actions';
+import { Project } from 'sharedTypes';
+import { updateProjectWithMedia } from 'renderer/util';
+import ModalTitle from '../ModalTitle';
+import SelectMediaBlock from '../SelectMediaBlock';
+import MediaDisplayOnUpload from '../MediaDisplayOnUpload';
+import StandardButton from '../StandardButton';
+
+interface Props {
+  prevView: () => void;
+  closeModal: () => void;
+  nextView: () => void;
+}
+
+const CustomStack = styled(Stack)`
+  width: 100%;
+`;
+
+const Container = styled(Box)`
+  background-color: ${colors.grey[700]};
+  ${(props) => props.theme.breakpoints.up('xs')} {
+    height: 500px;
+  }
+`;
+
+const UploadVideoView = ({ prevView, closeModal, nextView }: Props) => {
+  const [isAwaitingMedia, setIsAwaitingMedia] = useState<boolean>(true);
+  const [mediaFilePath, setMediaFilePath] = useState<string | null>(null);
+  const [mediaFileName, setMediaFileName] = useState<string | null>(null);
+  const currentProject = useSelector(
+    (store: ApplicationStore) => store.currentProject
+  );
+
+  const dispatch = useDispatch();
+
+  if (currentProject === null) {
+    return null;
+  }
+
+  const projectName = currentProject.name;
+
+  const setCurrentProject = (project: Project) =>
+    dispatch(projectCreated(project));
+  const addToRecentProjects = (project: Project) =>
+    dispatch(recentProjectAdded(project));
+
+  const handleTranscribe = async () => {
+    const project = await updateProjectWithMedia(currentProject, mediaFilePath);
+
+    if (project === null || mediaFilePath === null) {
+      return;
+    }
+
+    setCurrentProject(project);
+    addToRecentProjects(project);
+
+    nextView();
+  };
+
+  const transcribeButton = (
+    <StandardButton
+      color="primary"
+      onClick={handleTranscribe}
+      disabled={isAwaitingMedia}
+    >
+      Transcribe
+    </StandardButton>
+  );
+
+  const cancelButton = (
+    <StandardButton color="secondary" onClick={prevView}>
+      Back
+    </StandardButton>
+  );
+
+  return (
+    <Container>
+      <CustomStack
+        direction="column"
+        alignItems="flex-start"
+        justifyContent="space-between"
+        sx={{ height: '50%' }}
+      >
+        <CustomStack
+          direction="row"
```

Comment:
same as above :)

Reply:
done in: https://github.com/patrickbrett/mlvet/pull/38/commits/9c3bb86622a4c497c326922f4841c70ba1e791ac